### PR TITLE
Fix padding on rlp value

### PR DESCRIPTION
**Pull request by 0xSasaPrsic**
# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- [ ] Feature
- [x] Bugfix
- [ ] Refactor
- [ ] Format
- [ ] Documentation
- [ ] Testing
- [ ] Other:

## Description
Removes `H256::from_slice` which panics if a provided slice is not 32 bytes.
It appends leading zeros up to 32 bytes if the rlp value (message hash) has leading zeros that are trimmed.  
Prevents code from panicking and gets the correct hash (32 bytes) from the storage proof/account. 

## Checklist
- [x] I have performed a self-review of my own code.
- [x] The tests pass successfully with `cargo test`.
- [x] The code was formatted with `cargo fmt`.
- [ ] The code compiles with no new warnings with `cargo build --release` and `cargo build --release --features runtime-benchmarks`.
- [ ] The code has no new warnings when using `cargo clippy`.
- [ ] If this change affects documented features or needs new documentation, I have created a PR with a [documentation update](https://github.com/availproject/availproject.github.io).

### CODE REVIEW

**Review by luka-ethernal — APPROVED** [review]
LTGM

File: pallets/vector/src/storage_utils.rs [hunk 1]
```diff
@@ -81,19 +84,42 @@ pub fn get_storage_root(
...
 		Err(StorageError::StorageValueError)
 	}
 }
 
+/// padded_value it converts rlp value to H256 which requires 32 bytes and
+/// appends leading zeros if value is less than 32 bytes, otherwise throws an error.
+fn padded_value(rlp_value: &[u8]) -> Result<H256, StorageError> {
+	let mut slot_value = [0u8; 32].to_vec();
+
+	if rlp_value.len() < 32 {
```
Comment (ToufeeqP) [inline pallets/vector/src/storage_utils.rs:100]: may consider using a match here

**Review by ToufeeqP — APPROVED** [review]

File: pallets/vector/src/storage_utils.rs (hunk 1 above)
Comment (0xSasaPrsic) [reply pallets/vector/src/storage_utils.rs:100]: clippy has a suggestion for this but seems more complex than `if` `else`

Comment (ToufeeqP) [reply pallets/vector/src/storage_utils.rs:100]: Should be very similar current code, anyway, it's your call

**Review by Leouarz — APPROVED** [review]

**Review by markopoloparadox — APPROVED** [review]

**Pull request merged**